integration-tests/test_register.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+import functools
+import logging
+
+logger = logging.getLogger(__name__)
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_register(insights_client):
+    """
+    Test `insights-client --register` when the client is registered
+    """
+    response = insights_client.run("--register", check=False)
+    assert (
+        response.returncode == 0
+    ), "An application returns 0 after successful registration"
+    assert (
+        "Successfully registered host" in response.stdout
+    ), "An application should inform that a system was registered"
+    assert (
+        "Starting to collect Insights data" in response.stdout
+    ), "An application should inform that Insights data will be collected"
+    assert (
+        "Successfully uploaded report" in response.stdout
+    ), "An application should inform that report was successfully uploaded"
+    assert (
+        "View the Red Hat Insights console" in response.stdout
+    ), "An application should inform about Insights console"
+    assert (
+        insights_client.is_registered
+    ), "Current state of registration should be 'registered'"
```

Comment:
I am trying to highlight the main sense of a verification. It helps in case the verification fails. If a string was changed - the assertation message helps me to decide whether the verification should be changed (ie. the behaviour comply the main sense of the verification) or it is the right failure.

Reply:
what you describe can be also done without that: the assertion will be shown, and you get the code around it anyway

Reply:
the message next to assertation provides you better error message in pytest overall report (in case test Failed).